### ISSUE TITLE
Wrappers are breaking the purpose of the smart contract, no?

### ISSUE THREAD

**Issue by MBoui** [Web3-API/prototype#598]
Hi,

First all, i really like what you try to achieve.
I am looking to use a tools like `polywrap` for my next project.
I was wondering about the wrapper usage, which is something like a `middleman` between the `dapps` and the `smart contract`. 

One of the purpose of the smart contract and why this idea is so incredible. This is because, i can see the full code source and know what happening behind my `dapps`, right? I feel the wrapper concept broke this idea and add potentially an obfuscation behind my `dapps`. May be i am wrong, what do you think? Di you already spoke about this concern? And what was the conclusion.

**Comment by dOrgJelli** [comment]
Hi @MBoui, I'm not too sure how to approach answering your question, as it appears you're fairly new to web3 development (congrats and welcome :clap: :smile:).

I think there are a few fundamental understandings that will help guide you. Here are some brief intros + links to get you started...

**1. What are smart contracts, why are they useful, when should they be used?**
After a quick search, I found this really great 20 minute presentation that walks through [the what/why/how of smart-contracts](https://youtu.be/_J6G5g-nKg0). The purpose of this video is to help show you that smart contracts are much more than just transparent source code.

**2. How do you integrate smart contract protocols?**
Since smart contract protocols are often times complex due to limitations of on-chain execution, there is often times necessary computation that needs to occur within the dApp before interacting with the on-chain smart contracts. Most commonly today this is bundled into JavaScript SDKs that dApp developers can integrate into their dApps. Here are some example JS SDKs:
[Maker DAO (DAI)](https://docs.makerdao.com/build/dai.js/getting-started)  
[Uniswap](https://docs.uniswap.org/sdk/guides/quick-start)  
[Aave](https://docs.aave.com/developers/getting-started/sdks)  

**3. Why are Polywrappers better than traditional SDKs?**  
Here you can learn about [what Polywrap is](https://docs.polywrap.io/getting-started/what-is-polywrap) and why it presents a [better integration solution than traditional JavaScript SDKs](https://docs.polywrap.io/getting-started/polywrap-vs-javascript-sdks). Additionally I gave a [talk at EthCC](https://ethcc.polywrap.io/) that's pretty comprehensive, and might help guide your understanding as well.

**TL;DR**  
Polywrappers do not defeat the purpose of smart-contracts, but rather they enhance smart-contract protocols in ways that traditional JavaScript SDKs cannot.

**Comment by MBoui** [comment]
thanks for your answer, and yes i am new on the web3 development :)

**Comment by dOrgJelli** [comment]
Amazing, welcome once again @MBoui and god speed! Please don't hesitate to reach out via Discord, the Polywrap dev community is very friendly and eager to help newcomers building with the tool. NOTE: Polywrap is still very early dev tooling, but rapidly solidifying :smile:.
